File: zapp-common/src/main/java/com/zfoo/app/zapp/common/protocol/group/auth/SaveChannelAuthResponse.java
/*
 * Copyright (C) 2020 The zfoo Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and limitations under the License.
 */

package com.zfoo.app.zapp.common.protocol.group.auth;

import com.zfoo.protocol.IPacket;

/**
 * @author jaysunxiao
 * @version 1.0
 * @since 2020-04-23 14:20
 */
public class SaveChannelAuthResponse implements IPacket {

    public static final transient short PROTOCOL_ID = 18513;

    public static SaveChannelAuthResponse valueOf() {
        var response = new SaveChannelAuthResponse();
        return response;
    }

    @Override
    public short protocolId() {
        return PROTOCOL_ID;
    }

}

File: zapp-gateway/jsProtocol/group/auth/SaveChannelAuthResponse.js
// @author jaysunxiao
// @version 1.0
// @since 2020-04-23 14:20
const SaveChannelAuthResponse = function() {
};

SaveChannelAuthResponse.prototype.protocolId = function() {
    return 18513;
};

SaveChannelAuthResponse.write = function(buffer, packet) {
    if (buffer.writePacketFlag(packet)) {
        return;
    }
};

SaveChannelAuthResponse.read = function(buffer) {
    if (!buffer.readBoolean()) {
        return null;
    }
    const packet = new SaveChannelAuthResponse();
    return packet;
};

export default SaveChannelAuthResponse;
